core/blockchain.go

```diff
@@ -2183,18 +2183,12 @@ func (bc *BlockChain) CacheConfig() *CacheConfig {
 	return bc.cacheConfig
 }
 
-func (bc *BlockChain) setTxIndexTail(newTail uint64) error {
+func (bc *BlockChain) repairTxIndexTail(newTail uint64) error {
 	bc.txIndexTailLock.Lock()
 	defer bc.txIndexTailLock.Unlock()
 
-	tailP := rawdb.ReadTxIndexTail(bc.db)
-	var tailV uint64
-	if tailP != nil {
-		tailV = *tailP
-	}
-
-	if newTail > tailV {
-		log.Info("Repairing tx index tail", "old", tailV, "new", newTail)
+	if currentTail := rawdb.ReadTxIndexTail(bc.db); currentTail == nil || newTail > *currentTail {
```

Comment:
Question: as far as I can tell this is just a refactor with the same logic; is that your intention?

Comment: I think the variable `currentTail` could just be `curr` (or even `tail`) to make this more succinct and easier to read. The fact that it's a tail is implied by the RHS of the assignment, and it's only used on two lines afterwards so verbosity is unnecessary:

```go
if curr := rawdb.ReadTxIndexTail(bc.db); curr == nil || *curr < newTail {
// or
if tail := rawdb.ReadTxIndexTail(bc.db); tail == nil || *tail < newTail {
```

Afterthought: my using `curr < new` instead of `new > curr` just kinda happened as it felt more natural to frame everything in terms of `curr`. Entirely subjective, and up to you to decide if you think I'm nuts :)

Reply:
yes it's a refactor

I applied your suggestion, thanks.